kernel/src/scan/mod.rs

```diff
@@ -833,10 +802,25 @@ impl StateInfo {
             }
         }
 
+        let physical_schema = Arc::new(StructType::try_new(read_fields)?);
+
+        let physical_predicate = match predicate {
+            Some(pred) => PhysicalPredicate::try_new(&pred, &logical_schema)?,
+            None => PhysicalPredicate::None,
+        };
+
+        let transform_spec =
+            if !transform_spec.is_empty() || column_mapping_mode != ColumnMappingMode::None {
```

Comment:
If we're _only_ generating a transform because of column mapping, that's not _really_ a transform, it's more like a "schema change". I guess we could model that as some kind of `Identity` expression + a schema, or maybe there's a better way to do it, but regardless, generating a bunch of identity transforms per column is probably silly. 

not to be fixed in this PR, but can we make a follow-up issue?